slackbot/bot.py

```diff
@@ -37,6 +37,9 @@ def _keepactive(self):
         logger.info('keep active thread started')
         while True:
             time.sleep(30 * 60)
+            users = dict((u['id'], u) for u in self._client.webapi.users.list()['body']['members'])
+            if users != self._client.users:
+                self._client.users = users
```

Comment:
This looks good as a first-step fix. But i think we should also trigger a user list update immediately when the bot accept a request from a user which is not in the current user list.